server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -2242,9 +2242,10 @@ && isNotUnitVector(squaredMagnitude)) {
             int adjustedK = k;
             // By default utilize the quantized oversample is configured
             // allow the user provided at query time overwrite
-            Float oversample = indexOptions instanceof QuantizedIndexOptions quantizedIndexOptions
-                ? quantizedIndexOptions.rescoreVector != null ? quantizedIndexOptions.rescoreVector.oversample() : null
-                : null;
+            Float oversample = null;
+            if (indexOptions instanceof QuantizedIndexOptions quantizedIndexOptions && quantizedIndexOptions.rescoreVector != null) {
```

Comment:
Nit - Why go through this if branch if queryOversample is not null? It will be overriden in the next if.